gateway/api/serializers.py

```diff
@@ -24,23 +31,56 @@ class UploadProgramSerializer(serializers.ModelSerializer):
 
     entrypoint = serializers.CharField(required=False)
     image = serializers.CharField(required=False)
+    provider = serializers.CharField(required=False)
 
     class Meta:
         model = Program
 
-    def retrieve_one_by_title(self, title, author):
+    def separate_provider_from_title(self, title):
         """
-        This method returns a Program entry if it finds an entry searching by the title, if not None
+        This method returns provider and title from a title with /
         """
-        return (
-            Program.objects.filter(title=title, author=author)
-            .order_by("-created")
-            .first()
+        title_split = title.split("/")
+        if len(title_split) == 1:
+            return None, title_split[0]
+        return title_split[0], title_split[1]
+
+    def check_provider_access(self, provider_name, author):
+        """
+        This method check if the author has access to the provider
+        """
+        provider = Provider.objects.filter(name=provider_name).first()
+        if provider is None:
+            logger.error("Provider [%s] does not exist.", provider_name)
+            return False
+        logger.error(
```

Comment:
is there a missing check here? I think this will log an error every time this function is run...

Reply:
Good catch 🙏 , solved in: [edb541c](https://github.com/Qiskit/qiskit-serverless/pull/1331/commits/edb541c03fa80f2062da29da509590945c20f967)